File: src/main/java/aurocosh/divinefavor/common/muliblock/common/PatchouliMultiblockAdapter.java
package aurocosh.divinefavor.common.muliblock.common;

import aurocosh.divinefavor.common.core.ResourceNamer;
import aurocosh.divinefavor.common.muliblock.ModMultiBlock;
import aurocosh.divinefavor.common.muliblock.MultiBlockConfiguration;
import aurocosh.divinefavor.common.muliblock.patchouli.PatchouliMultiBlockData;
import net.minecraft.util.ResourceLocation;
import vazkii.patchouli.api.IMultiblock;
import vazkii.patchouli.api.PatchouliAPI;

public class PatchouliMultiblockAdapter {
    public static void register(ModMultiBlock multiBlock) {
        for (MultiBlockConfiguration configuration : multiBlock.configurations) {
            if (!configuration.primary)
                continue;
            PatchouliMultiBlockData multiBlockData = new PatchouliMultiBlockData(configuration);
            System.out.print(multiBlockData.toString());
            System.out.println();

            IMultiblock iMultiblock = PatchouliAPI.instance.makeMultiblock(multiBlockData.pattern, multiBlockData.matchingData);
            ResourceLocation location = ResourceNamer.getFullName("patchouli", configuration.name);
            PatchouliAPI.instance.registerMultiblock(location, iMultiblock);
        }
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/muliblock/common/PatchouliMultiblockAdapter.kt
package aurocosh.divinefavor.common.muliblock.common

import aurocosh.divinefavor.common.core.ResourceNamer
import aurocosh.divinefavor.common.muliblock.ModMultiBlock
import aurocosh.divinefavor.common.muliblock.patchouli.PatchouliMultiBlockData
import vazkii.patchouli.api.PatchouliAPI

object PatchouliMultiblockAdapter {
    fun register(multiBlock: ModMultiBlock) {
        for (configuration in multiBlock.configurations) {
            if (!configuration.primary)
                continue
            val multiBlockData = PatchouliMultiBlockData(configuration)

            print(multiBlockData.toString())
            println()

            val iMultiblock = PatchouliAPI.instance.makeMultiblock(multiBlockData.pattern, *multiBlockData.matchingData)
            val location = ResourceNamer.getFullName("patchouli", configuration.name)
            PatchouliAPI.instance.registerMultiblock(location, iMultiblock)
        }
    }
}
